examples/riot/external_modules/reactor-uc/Makefile

```diff
@@ -0,0 +1,4 @@
+include $(RIOTBASE)/Makefile.base
+
+all:
+	$(QQ)"$(MAKE)" -C ../../../../src -f $(CURDIR)/reactor-uc.mk
```

Comment:
I personally prefer not to have so much platform specific files in the root folder, but if you think this is fine then sure we can move the make files for riot into the root folder. This can then also be reused in package of reactor-uc that is included into RIOT OS in @tanneberger's fork :)

Reply:
Okay then I would propose to put RIOT as a submodule into `external/platform/RIOT` for now.

> we need to add nanopb sources to the build. This file can be used by the RIOT build system, or by another Makefile-based build system if the user wants to integrate it with something different.

Yes this seems reasonable I will see to move the non-riot-specific parts to the root folder.

> I was really struggling with getting the RIOT make system to do what I wanted

The same for me :) The RIOT build system can be quite tough and confusing. I myself are still quite confused about how exactly that Makefile.include file exactly works... I hope I can make the reactor-uc external module better soon.